design/observers.md

```diff
@@ -0,0 +1,211 @@
+ There will probably be a gap between the latest Validators state and latest Observer state.
+    - It means that clients (indy-sdk) need to have either a separate pool ledger for each Observer, or a Pool State trie (Patricia trie)
+- There is enough Gatekeeper Nodes to create a separate Ring
+
+###Iterations
+It’s better to implement the long-term solution in multiple iterations (taken into account the Priority of Goals above):
+1. Support Gatekeepers only (that is trusted Observers which are always in sync)
+1. Support Observers (untrusted Observers which may not be 100% in sync)
+1. “Close the gate” to Validators to allow Validators be connected to Gatekeepers only
+
+Note: We may change the order of Iterations (and hence the roadmap of tasks in the last section) depending on the requirements and priority of Goals
+
+###Data to keep in sync on Gatekeepers and Observers
+- All ledgers
+- All states
+- Attributes store
+- BLS store (in fact the last entry only) 
+- seqNo store
+
+###Participants
+
+##### Catch-up
+|  | What ledger | Min Number of nodes to finish | Node type to finish | Bootstrapping 
+| --- | --- | --- | --- | --- 
+Client | POOL | F_G + 1 (or even 1 if use state proofs and timestamp)| Gatekeepers | Yes (from Observers and Gatekeepers)
+Observer | All | F_G + 1 (or even 1 if use state proofs and timestamp)| Gatekeepers | Yes (from Observers and Gatekeepers)
+Gatekeeper | All | F_V + 1 (or even 1 if use state proofs and timestamp)| Validators | Yes (from Gatekeepers and Validators)
+Validator | All | F_V + 1 (or even 1 if use state proofs and timestamp)| Validators | Yes (from Gatekeepers and Validators)
+
+##### Read and Write
+| | Read Node type | Read Min Number of nodes | Write Node type | Write Min Number of nodes | 
+| --- | --- | --- | --- | ---  |
+Client | 1 | Observer OR Gatekeeper | 1 (eventually propagated to f_V+1 validators by Observers and Gatekeepers) | Observer OR Gatekeeper
+
+##### Keep in sync
+
+| | How | Node type to connect to for sync
+| --- | --- | ---
+Client | No (do catch-up for syncing); TBD in future |  
+Observer | Catch-up + Any custom policy | Other Observers and Gatekeepers
+Gatekeeper | Catch-up + Receive all write Replies | Validators
+Validator | Catch-up | Validators 
+
+#### Description
+- Validators Ring: 
+    - Has its own Steward, and added to the Pool by a NODE txn with VALIDATOR service
+    - Participates in consensus (write requests ordering)
+    - Do not process any Clients nor Observers requests
+    - Supports restricted connections to other Validators and Gatekeepers only (not to Observers)
+    - Provides catch-up for Gatekeepers and other Validators only (neither for Clients nor other Observers)
+    - Connected to all Gatekeepers (the same way as to Validators, but just doesn’t include them into 3PC)
+    - Propagates all write replies to all Gatekeepers to make them in sync
```

Comment:
Do all validators tell all gatekeepers, cant some "special validator" say the primary tell all since it knows the correct BLS earliest, each gatekeeper can independently ask other non-primary validators or other gatekeepers for any changes not communicated by primary. The "special validator" does not  have to be primary.

Reply:
I believe we should start with the simplest working option: all Validators tell all Gatekeepers. 
Later on we can evolve it.